### PR TITLE
Java JNI for Multiple contains

### PR DESCRIPTION
**Pull request by res-life**
## Description
This is Java JNI interface for [multiple contains PR](https://github.com/rapidsai/cudf/pull/16900)

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 1]
```diff
@@ -3332,6 +3332,44 @@ public final ColumnVector stringContains(Scalar compString) {
     return new ColumnVector(stringContains(getNativeView(), compString.getScalarHandle()));
   }
 
+  private static long[] toPrimitive(Long[] longs) {
```
Comment (abellina) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:3335]: unused?

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 2]
```diff
@@ -3332,6 +3332,44 @@ public final ColumnVector stringContains(Scalar compString) {
...
+   *   column 0: [true, false, false]
+   *   column 1: [false, false, true]
+   *
+   * @param targets UTF-8 encoded strings to search for in each string in `input`
+   * @return BOOL8 columns
+   */
+  public final ColumnVector[] stringContains(ColumnView targets) {
+    assert type.equals(DType.STRING) : "column type must be a String";
+    assert targets.getType().equals(DType.STRING) : "targets type must be a string";
+    assert targets.getNullCount() == 0 : "targets must not be null";
```
Comment (abellina) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:3367]:
```suggestion
    assert targets.getNullCount() == 0 : "targets must not contain nulls";
```

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 3]
```diff
@@ -4437,6 +4475,13 @@ private static native long stringReplaceWithBackrefs(long columnView, String pat
    */
   private static native long stringContains(long cudfViewHandle, long compString) throws CudfException;
 
+  /**
+   * Native method for searching for the given target strings within each string in the provided column.
+   * @param cudfViewHandle native handle of the cudf::column_view being operated on.
+   * @param targets handle of the column containing the string being searched for.
```
Comment (abellina) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:4481]:
```suggestion
   * @param targets handle of the column view containing the string being searched for.
```

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 4]
```diff
@@ -4437,6 +4475,13 @@ private static native long stringReplaceWithBackrefs(long columnView, String pat
...
+   */
+  private static native long[] stringContainsMulti(long cudfViewHandle, long targets) throws CudfException;
```
Comment (abellina) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:4483]:
```suggestion
  private static native long[] stringContainsMulti(long cudfViewHandle, long targetViewHandle) throws CudfException;
```

File: java/src/main/native/src/ColumnViewJni.cpp [hunk 5]
```diff
@@ -2827,4 +2828,25 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_toHex(JNIEnv* env, jclass
   }
   CATCH_STD(env, 0);
 }
+
+JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_ColumnView_stringContainsMulti(JNIEnv* env,
+                                                                                jobject j_object,
+                                                                                jlong j_view_handle,
+                                                                                jlong comp_strings)
```
Comment (abellina) [inline java/src/main/native/src/ColumnViewJni.cpp:2835]:
```suggestion
                                                                                jlong j_target_view_handle)
```

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 6]
```diff
@@ -3828,6 +3828,30 @@ void testStringOpsEmpty() {
...
+      ColumnVector[] results = null;
+      try (ColumnVector haystack = ColumnVector.fromStrings("All the leaves are brown",
+          "And the sky is grey",
+          "I've been for a walk",
+          "On a winter's day",
+          null,
+          "");
+          ColumnVector targets = ColumnVector.fromStrings("the", "a");
+           ColumnVector expected0 = ColumnVector.fromBoxedBooleans(true, true, false, false, null, false);
+           ColumnVector expected1 = ColumnVector.fromBoxedBooleans(true, false, true, true, null, false)) {
```
Comment (abellina) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:3842]:
```suggestion
          ColumnVector expected0 = ColumnVector.fromBoxedBooleans(true, true, false, false, null, false);
          ColumnVector expected1 = ColumnVector.fromBoxedBooleans(true, false, true, true, null, false)) {
```

File: java/src/main/java/ai/rapids/cudf/ColumnView.java (hunk 2 above)
Comment (res-life) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java]: done.

File: java/src/main/java/ai/rapids/cudf/ColumnView.java (hunk 3 above)
Comment (res-life) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java]: done.

File: java/src/main/java/ai/rapids/cudf/ColumnView.java (hunk 4 above)
Comment (res-life) [reply java/src/main/java/ai/rapids/cudf/ColumnView.java]: done.

File: java/src/main/native/src/ColumnViewJni.cpp (hunk 5 above)
Comment (res-life) [reply java/src/main/native/src/ColumnViewJni.cpp]: done.

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java (hunk 6 above)
Comment (res-life) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java]: done.

**Review by abellina — APPROVED** [review]
LGTM

**Review by revans2 — APPROVED** [review]

**Pull request merged**